jobs/V3/basic_job_sample.rb

```diff
@@ -12,213 +12,228 @@
+		update_job_request.job = job_to_be_updated;
+		update_job_request.update_mask = field_mask;
+		job_updated= talentSolution_client.patch_project_job(job_name, update_job_request);
+		puts "Job updated with filedMask " + update_job_request.update_mask + ". Updated job: " + job_updated.to_json;
+		return job_updated;
+	rescue
+		puts "Got exception while updating job with fieldMask"
+		splitted_name = job_name.split('/');
+		if splitted_name[0] != "projects" || splitted_name[2] != "jobs" || splitted_name[1].empty? || splitted_name[3].empty?
+			puts "Invalid jobName format";
+		elsif get_job(job_name).nil?
+			puts "job doesn't exist";
 		end
 	end
-# [END update_job]
-
-# [START update_job_with_field_mask]
-=begin 
-		Update a job with field mask
-=end
-	def updateJobWithFieldMask(jobName, fieldMask, jobToBeUpdated)
-		begin
-			updateJobRequest = @@Jobs::UpdateJobRequest.new;
-			updateJobRequest.job = jobToBeUpdated;
-			updateJobRequest.update_mask = fieldMask;
-			jobUpdated= @@talentSolution_client.patch_project_job(jobName, updateJobRequest);
-			puts "Job updated with filedMask " + updateJobRequest.update_mask + ". Updated job: " + jobUpdated.to_json;
-			return jobUpdated;
-		rescue
-			puts "Got exception while updating job with fieldMask"
-			splitted_name = jobName.split('/');
-			if splitted_name[0] != "projects" || splitted_name[2] != "jobs" || splitted_name[1].empty? || splitted_name[3].empty?
-				puts "Invalid jobName format";
-			elsif getJob(jobName).nil?
-				puts "job doesn't exist";
-			end
+	# [END update_job_with_field_mask]
+end
+
+def delete_job(job_name)
+	# [START delete_job]
+	require "google/apis/jobs_v3"
+
+	jobs   = Google::Apis::JobsV3;
+
+	talentSolution_client = jobs::CloudTalentSolutionService.new;
+	talentSolution_client.authorization = Google::Auth.get_application_default(
+	"https://www.googleapis.com/auth/jobs"
+	)
+
+	begin
+		talentSolution_client.delete_project_job(job_name);
+		puts "Job deleted. jobName: " + job_name;
+	rescue
+		puts "Got exception while deleting job"
+		splitted_name = job_name.split('/');
+		if splitted_name[0] != "projects" || splitted_name[2] != "jobs" || splitted_name[1].empty? || splitted_name[3].empty?
+		puts "Invalid jobName format";
+		elsif get_job(job_name).nil?
+		puts "job doesn't exist";
 		end
 	end
-# [END update_job_with_field_mask]
-
-# [START delete_job]
-=begin 
-		Delete a job
-=end
-	def deleteJob(jobName)
-		begin
-			@@talentSolution_client.delete_project_job(jobName);
-			puts "Job deleted. jobName: " + jobName;
-		rescue
-			puts "Got exception while deleting job"
-			splitted_name = jobName.split('/');
-			if splitted_name[0] != "projects" || splitted_name[2] != "jobs" || splitted_name[1].empty? || splitted_name[3].empty?
-				puts "Invalid jobName format";
-			elsif getJob(jobName).nil?
-				puts "job doesn't exist";
-			end
-		end
+	# [END delete_job]
+end
+
+def run_basic_job_sample arguments
+
+	require_relative "basic_company_sample"
```

Comment:
this line isn't required.

Reply:
I think it is required... I am using the functions in basic_company_sample to generate the company before creating jobs. If I didn't have this line, I can not refer to that function. Pls correct me if I was wrong.

Reply:
Ah sorry my mistake. You're correct that you need this to do that.

Why are you generating the company before creating the jobs? I would expect a CLI to create companies and another separate CLI to create jobs based on the created companies. At that point it becomes more of an end-user task to provide the information.

Reply:
Make sense. What about this: I take company_name as user input when they create job, and if the company exists, I create job under that company; if not I create the company for them before creating job? What do you think?

Reply:
The samples should focus on what the API should demonstrate to a developer:
Task one: Create a new company. **Error checking** to make sure the company doesn't exist yet is okay here.
Task two: Create a new job with a company name that should exist. **Error checking** to make sure the Company exists is fine but do not create a new company in this sample.